src/main/java/com/redhat/devtools/intellij/tektoncd/ui/wizard/ParametersStep.java

```diff
@@ -121,20 +127,38 @@ public void setContent() {
 
     private void addListener(String idParam, JTextField txtValueParam, Border defaultBorder, int row) {
         // listen to when the focus is lost by the textbox and update the model so the preview shows the updated value
-        txtValueParam.addFocusListener(new FocusAdapter() {
+        txtValueParam.getDocument().addDocumentListener(new DocumentListener() {
             @Override
-            public void focusLost(FocusEvent e) {
-                super.focusLost(e);
-                setInputValue(model.getParams(), idParam, txtValueParam.getText());
-                // reset error graphics if an error occurred earlier
-                if (isValid(txtValueParam)) {
-                    txtValueParam.setBorder(defaultBorder);
-                    if (errorFieldsByRow.containsKey(row)) {
-                        deleteComponent(errorFieldsByRow.get(row));
-                    }
+            public void insertUpdate(DocumentEvent e) {
+                update();
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+                update();
+            }
 
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+                update();
+            }
+
+            public void update() {
+                if (scheduler != null && !scheduler.isCancelled() && !scheduler.isDone()) {
+                    scheduler.cancel(true);
                 }
-                fireStateChanged();
+
+                scheduler = executor.schedule(() -> {
```

Comment:
I used the scheduler to prevent to make tens of update while the user is still typing. When the user stops for a while (500 millisecond) i force a preview update. Otherwise i'm going to update the preview panel for each single letter typed by the user. Are there other alternatives?

Reply:
No but this is required only if:
- computation of the update is costly
- UI glitch is observed when typing

Reply:
Removed as you said. It is useless.